travelrec/recommendations_pipeline.py

```diff
@@ -32,19 +44,31 @@ def recommendations_pipeline(query, nlp, verbose=False):
     location_entities = processed_statement.location_entities()
     logging.info(f"Locations: {location_entities}")
 
-    loc = location_entities[0].text
-    latitude, longitude = get_location(loc)
-    logging.info(f"Coordinates: {loc} - ({latitude}, {longitude})")
+    if len(location_entities) == 0:
+        raise NoLocationsFoundError
+
+    location_coordinates = [get_location(loc.text) for loc in location_entities]
+    logging.info("Coordinates:")
+    for loc, coord in zip(location_entities, location_coordinates):
+        logging.info(f"\t{loc} - ({coord[0]}, {coord[1]})")
 
     # climate filters
     climate_predicates = climateSim.construct_filters(processed_statement.climate_terms())
-    climate_predicate = climate_predicates[0]
-    logging.info(f"Climate: {climate_predicate}")
-    nearby_cities = get_cities_with_temperature(
-        latitude, longitude, search_distances['climate'], climate_predicate
-    )
+    logging.info(f"Climate: {climate_predicates}")
+    nearby_cities = []
+    for coord in location_coordinates:
+        nearby_cities.extend(get_cities_with_temperature(
```

Comment:
For locations I would raise error if more than one named entity is detected. Otherwise, the results are going to be intertwined and unreadable.

Reply:
done